### ISSUE TITLE
`consistent-arrow-return-style`: option to work around implicit return when the entire statement exceeds `max-len`

### ISSUE THREAD

**Issue by ShayanTheNerd** [sindresorhus/eslint-plugin-unicorn#3604]
### Description

When using this rule together with ESLint Stylistic's [`@stylistic/max-len`](https://eslint.style/rules/max-len), the preferred arrow return style can conflict with the configured maximum line length.
For example:
```js
{
  '@stylistic/max-len': ['warn', {
    tabWidth: 2,
    code: 120,
    ignoreUrls: true,
    ignoreComments: true,
    ignoreTrailingComments: true,
  }],
}
```
A long implicit-return statement often cannot be wrapped naturally without violating the configured maximum line length:
```js
const veryLongVariableName = anotherLongVariableName.map((longItemName) => `SuperLongTemplateLiteralWith${longItemName}`);
```
Using an explicit return allows the returned expression to be placed on its own line, making it compatible with `max-len` while preserving readability.

Please add an option to ignore the implicit return style when the entire implicit-return statement would exceed the configured `max-len` limit, or preferably enforce an explicit `return` instead.

### Examples

When the entire implicit-return statement fits within `max-len`:
```js
// ❌
const shortVarName = anotherShortVarName.map((item) => {
  return `With${item}`
});

// ✅
const shortVarName = anotherShortVarName.map((item) => `With${item}`);
```

When the entire implicit-return statement exceeds `max-len`:
```js
// ❌
const veryLongVariableName = anotherLongVariableName.map((longItemName) => `SuperLongTemplateLiteralWith${longItemName}`);

// ✅
const veryLongVariableName = anotherLongVariableName.map((longItemName) => {
  return `SuperLongTemplateLiteralWith${longItemName}`;
});
```

**Comment by sindresorhus** [comment]
`consistent-arrow-return-style` intentionally makes a structural decision based on whether the arrow body is multiline. It does not (and should not) inspect other rules. `@stylistic/max-len` is line based and supports options for tabs, comments, strings, templates, URLs, and patterns. Mirroring that configuration here would duplicate another rule's policy and still produce mismatches.

In most cases, the implicit return can stay in place by wrapping the surrounding expression:

```js
const values = veryLongVariableName.map(
	(longItemName) => `SuperLongTemplateLiteralWith${longItemName}`,
);
```

For cases without a suitable wrapping point, please use a targeted disable.

I'm closing this as out of scope.
